message/name.go

```diff
@@ -0,0 +1,179 @@
+package message
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/dogmatiq/dogma"
+)
+
+// Name is the fully-qualified name of a message type.
+type Name struct {
+	n string
+}
+
+// NameOf returns the fully-qualified type name of v.
+func NameOf(v dogma.Message) Name {
+	rt := reflect.TypeOf(v)
+	n := buildName(rt, true)
+	return Name{n}
+}
+
+// String returns the fully-qualified type name as a string.
+func (n Name) String() string {
+	return n.n
+}
+
+// IsZero returns true if n is the zero-value.
+func (n Name) IsZero() bool {
+	return n.n == ""
+}
+
+// MarshalText returns a UTF-8 representation of the name.
+func (n Name) MarshalText() ([]byte, error) {
+	if n.n == "" {
+		return nil, fmt.Errorf("can not marshal empty name")
+	}
+
+	return []byte(n.n), nil
+}
+
+// UnmarshalText unmarshals a name from its UTF-8 representation.
+func (n *Name) UnmarshalText(text []byte) error {
+	if len(text) == 0 {
+		return fmt.Errorf("can not unmarshal empty name")
+	}
+
+	n.n = string(text)
+	return nil
+}
+
+// MarshalBinary returns a binary representation of the handler type.
```

Comment:
Should this doc comment be mentioning "handler type"?

Reply:
```suggestion
// MarshalBinary returns a binary representation of the name.
```